### PR TITLE
upcoming: [M3-7690] - Support PlacementGroups in CLI tool

### PR DESCRIPTION
**Pull request by carrillo-erik**
## Description 📝
This PR allows Placement Group data to be passed in as arguments via the `Linode CLI`. 

## Changes  🔄
- The Placement Group object is flattened to obtain and pass the `id` value. In the end the relevant argument and value are passed as follows: `placement_group.id 1234`.

## Target release date 🗓️
05/13/2024

## Preview 📷
| Before  | After   |
| ------- | ------- |
| Screenshot 2024-05-03 at 10 22 09 AM | linode-cli-arg |

### Prerequisites
(How to setup test environment)
- ...

### Verification steps
(How to verify changes)
- Verify that a new Linode instance can be created when passing a Placement Group id argument using the `linode-cli` tool.

## As an Author I have considered 🤔
*Check all that apply*

- [x] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/utilities/generate-cli.ts [hunk 1]
```diff
@@ -89,6 +92,12 @@ const dataEntriesReduce = (acc: string[], [key, value]: JSONFieldToArray) => {
   if (key === 'metadata') {
     const userData = value as UserData;
     acc.push(`  --${key}.user_data="${userData.user_data}"`);
+  } else if (key === 'placement_group') {
+    const placementGroup = value as PlacementGroup;
+    acc.push(`  --${key}.id ${placementGroup.id}`);
```
Comment (carrillo-erik) [inline packages/manager/src/utilities/generate-cli.ts:97]: This line could also be written as 
```
acc.push(`  --${key}.id="${placementGroup.id}"`);
```
Will need to check with DX team to ensure the correct formatting is applied to the `placement_group.id` argument.

Screenshot 2024-05-03 at 3 36 51 AM

Screenshot 2024-05-03 at 3 39 08 AM

File: packages/manager/src/utilities/generate-cli.test.ts [hunk 2]
```diff
@@ -43,6 +46,7 @@ const linodeDataForCLI = `
   --authorized_users Gritty \\
   --interfaces.ipam_address null --interfaces.ipv4.nat_1_1 \"any\" --interfaces.ipv4.vpc \"123\" --interfaces.label null --interfaces.primary true --interfaces.purpose \"vpc\" --interfaces.subnet_id 8296 \\
   --metadata.user_data="cmVrbmpnYmloZXVma2xkbQpqZXZia2Y=" \\
+  --placement_group.id 1234 \\
```
Comment (carrillo-erik) [inline packages/manager/src/utilities/generate-cli.test.ts:49]: This is tied to the comment on the other comment below, `placement_group.id` is expected to be an `int` and should not require `" "` marks around the value. Only `string` type values take `" "` marks.

**Comment by abailly-akamai** [comment]
we need a review from @lgarber-akamai

**Review by lgarber-akamai — APPROVED** [review]
LGTM from a CLI perspective!

**Review by dwiley-akamai — APPROVED** [review]

**Review by abailly-akamai — APPROVED** [review]

**Pull request merged**